core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -296,7 +296,20 @@ public static InetSocketAddress getBindAddress(ServiceType service) {
   }
 
   /**
-   * Gets a local host name for the host this JVM is running on.
+   * Gets the local hostname to be used by the client. If this isn't configured, a non-loopback
+   * local hostname will be looked up.
+   *
+   * @return the local hostname for the client
+   */
+  public static String getClientHostName() {
+    if (Configuration.containsKey(PropertyKey.USER_HOSTNAME)) {
+      return Configuration.get(PropertyKey.USER_HOSTNAME);
+    }
+    return getLocalHostName();
```

Comment:
Things that still use it:
- ApplicationMaster for the hostname to use for registering with YARN
- Metrics system for assigning an id to metrics
- A few dozen tests

I think the distinction is that when we're in the client, worker, or master we use their respective hostname lookups, otherwise we use `getLocalHostName()`.